internal/pkg/deploy/cloudformation/app.go

```diff
@@ -104,36 +103,40 @@ func (cf CloudFormation) upgradeAppStackSet(config *stack.AppStackConfig) error
 	}
 }
 
-func (cf CloudFormation) upgradeAppStack(s *cloudformation.Stack) error {
-	for {
-		// Upgrade app stack.
+func newRenderAppStackInput(cfnStack *cloudformation.Stack) *executeAndRenderChangeSetInput {
+	return &executeAndRenderChangeSetInput{
+		stackName:        cfnStack.Name,
+		stackDescription: fmt.Sprintf("Creating the infrastructure for the %s app.", cfnStack.Name),
+	}
+}
+
+func (cf CloudFormation) upgradeAppStack(conf *stack.AppStackConfig) error {
+	s, err := toStack(conf)
+	if err != nil {
+		return err
+	}
+	in := newRenderAppStackInput(s)
+	in.createChangeSet = func() (changeSetID string, err error) {
+		spinner := progress.NewSpinner(cf.console)
+		label := fmt.Sprintf("Proposing infrastructure changes for %s.", s.Name)
+		spinner.Start(label)
+		defer stopSpinner(spinner, err, label)
+
+		// get tags from existing stack
 		descr, err := cf.cfnClient.Describe(s.Name)
 		if err != nil {
-			return fmt.Errorf("describe stack %s: %w", s.Name, err)
-		}
-		if cloudformation.StackStatus(aws.StringValue(descr.StackStatus)).InProgress() {
```

Comment:
I wonder why we needed the retry - is this because we did automatic `app upgrade` from `svc deploy` before @iamhopaul123 do you remember?

I just wanted to make sure that we are removing codes that are actually removable 👀

Reply:
Yeah i guess that's the main reason. Previously `app upgrade` was called much more frequently then it is now, and we expected there would be very likely two upgrade happening at the same time will cause this issue without retrying. I seems ok to me to remove it now.